src/cache/inventory.ts

```diff
@@ -0,0 +1,109 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import * as cache from '@actions/cache';
+import * as core from '@actions/core';
+import * as glob from '@actions/glob';
+
+const INVENTORY_CACHE_KEY = 'volta-inventory';
+
+function getInventoryPath(voltaHome: string): string {
+    return path.join(voltaHome, 'tools', 'inventory');
+}
+
+function getImagePath(voltaHome: string): string {
+    return path.join(voltaHome, 'tools', 'image');
+}
+
+export async function restoreInventory(voltaHome: string): Promise<void> {
+    core.debug('Trying to restore inventory.');
+    const paths = [getInventoryPath(voltaHome)];
+    const cacheKey = await cache.restoreCache(paths, INVENTORY_CACHE_KEY);
+    if (cacheKey != null) {
+        core.info('Successfully restored inventory.');
+    } else {
+        core.info('Could not restore inventory.');
+    }
+}
+
+interface Tool {
+    name: string;
+    version: string;
+}
+
+function collectInstalledTools(voltaHome: string): Tool[] {
+    const imagePath = getImagePath(voltaHome);
+    return fs
+        .readdirSync(imagePath, { withFileTypes: true })
+        .filter((entry) => entry.isDirectory())
+        .flatMap((entry) =>
+            fs
+                .readdirSync(path.join(imagePath, entry.name), { withFileTypes: true })
+                .filter((subEntry) => subEntry.isDirectory())
+                .map(
+                    (subEntry) =>
+                        ({
+                            name: entry.name,
+                            version: subEntry.name,
+                        } satisfies Tool),
+                ),
+        );
+}
+
+function createToolFilePatter(tool: Tool): string {
+    switch (tool.name) {
+        case 'node':
+        case 'yarn':
+            return `${ tool.name }-v${ tool.version }*`;
+        default:
+            return `${ tool.name }-${ tool.version }*`;
+    }
+}
+
+async function cleanInventory(voltaHome: string, installedTools: Tool[]): Promise<void> {
```

Comment:
I don't fully grok how this is going to work to actually prune the inventory. If we call `restoreInventory` in the setup flow of the main action all of the tools & inventory will be copied over from prior runs, then (eventually) call `cleanInventory` in the `post` script for the action  -- how will we ever prune things out?

Can you walk me through the high level details?